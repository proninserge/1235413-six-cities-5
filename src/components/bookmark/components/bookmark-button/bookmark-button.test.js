import renderer from 'react-test-renderer';
import BookmarkButton from './bookmark-button';

const noop = () => {};

describe(`<BookmarkButton /> render`, () => {

  it(`correctly renders Bookmark button if favorite`, () => {
    const tree = renderer
      .create(
          <BookmarkButton
            className={`main`}
            onClick={noop}
            isFavorite={true}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`correctly renders Bookmark button if not favorite`, () => {
    const tree = renderer
      .create(
          <BookmarkButton
            className={`main`}
            onClick={noop}
            isFavorite={false}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});

