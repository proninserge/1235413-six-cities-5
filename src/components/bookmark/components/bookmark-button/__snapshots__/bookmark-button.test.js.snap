// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookmarkButton /> render correctly renders Bookmark button if favorite 1`] = `
<button
  className="main-button button main-button--active"
  onClick={[Function]}
  type="button"
>
  <svg
    className="main-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`<BookmarkButton /> render correctly renders Bookmark button if not favorite 1`] = `
<button
  className="main-button button "
  onClick={[Function]}
  type="button"
>
  <svg
    className="main-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
